icaruscode/Decode/MCDecoderICARUSTPCwROI_module.cc

```diff
@@ -498,110 +499,54 @@ void MCDecoderICARUSTPCwROI::processSingleLabel(art::Event&
-//
-//                // Ok, now store things...
-//                unsigned int planeIndex = fPlaneToROPPlaneMap.find(planeID)->second;
-//                unsigned int wire       = channel - fPlaneToWireOffsetMap.find(planeID)->second;
-//
-//                if (wire >= channelArrayPairVec[planeIndex].second.size()) continue;
-//
-//                icarus_signal_processing::VectorFloat& dataVec = channelArrayPairVec[planeIndex].second[wire];
-//
-//                for(size_t tick = 0; tick < dataSize; tick++) dataVec[tick] = rawDataVec[tick];
-//
-//                // Keep track of the channel
-//                channelArrayPairVec[planeIndex].first[wire] = daq::INoiseFilter::ChannelPlanePair(channel,planeID.Plane);
-            }
-        }
-
-        // Some detector simulations don't output channels that don't have any possibility of signal (ghost channels)
-        // Do a cleanup phase here to find these
-        for(auto& boardInfo : boardToChannelArrayPairMap)
-        {
-            if (boardWireCountMap[boardInfo.first] < 64)
-            {
-                processSingleImage(clockData, boardInfo.second, boardWireCountMap[boardInfo.first], concurrentRawDigits, concurrentRawRawDigits, coherentRawDigits, concurrentROIs);
-            }
-        }
+	  unsigned int board = channelToBoardItr->second.first;
+	  auto mapIter = boardToRawDigitMap.find(board);
+	  if (mapIter != boardToRawDigitMap.end()) {
+	    mapIter->second.push_back(rawDigit);
+	  } else {
+	    boardToRawDigitMap.insert({board,std::vector<const raw::RawDigit*>{rawDigit}});
+	  }
+	}
+	//std::cout << "boardToRawDigitMap.size()=" << boardToRawDigitMap.size() << std::endl;
+	//for (auto elem : boardToRawDigitMap) {
+	tbb::parallel_for (static_cast<std::size_t>(0),boardToRawDigitMap.size(),[&](size_t& r) {
+	  auto elem = std::next(boardToRawDigitMap.begin(),r);
+
+	  //std::cout << "board=" << elem->first << " nch=" << elem->second.size() << " first=" << elem->second[0]->Channel() << " last=" << elem->second[elem->second.size()-1]->Channel() << std::endl;
+
+	  std::vector<const raw::RawDigit*>& rawDigitVec = elem->second;
+	  // Sort (use a lambda to sort by channel id)
+	  std::sort(rawDigitVec.begin(),rawDigitVec.end(),[](const raw::RawDigit* left, const raw::RawDigit* right) {return left->Channel() < right->Channel();});
+          ChannelArrayPair chanArr;
+          for (const auto rawDigit : rawDigitVec) {
+	    // Declare a temporary digit holder and resize it if downsizing the waveform
+	    raw::RawDigit::ADCvector_t rawDataVec(dataSize);
+	    // Decompress data into local holder
+	    raw::Uncompress(rawDigit->ADCs(), rawDataVec, rawDigit->Compression());
+	    // Fill into the data structure
+            raw::ChannelID_t channel = rawDigit->Channel();
+            unsigned int planeIdx       = fChannelToBoardWirePlaneMap.find(channel)->second.second.second;
+	    icarus_signal_processing::VectorFloat boardDataVec(dataSize);
```

Comment:
It looks like this initialization + conversion + copy can be compacted into:
```suggestion
	    icarus_signal_processing::VectorFloat boardDataVec(rawDataVec.cbegin(), rawDataVec.cend());
```

Reply:
thanks